http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/YamlProvider.java

```diff
@@ -0,0 +1,66 @@
+package io.quarkus.ts.http.advanced.reactive;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.ext.MessageBodyReader;
+import jakarta.ws.rs.ext.MessageBodyWriter;
+import jakarta.ws.rs.ext.Provider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import io.quarkus.logging.Log;
+
+@Provider
+@Consumes("application/yaml")
```

Comment:
interesting, so given you didn't specify `@Produces`  this will be applied to all content types? Are you sure this is never applied to `public CityListDTO getCities() {`?

Reply:
Well, it will be applied to all resource methods that consume "application/yaml".
If a method resource @produce another different content type , it takes precedence over the general provider.
In the case that you mentioned,it is:
```
@Produces(value = MediaType.APPLICATION_XML)
    public CityListDTO getCities() {
```

Reply:
The `MessageBodyWriter` javadoc only speaks about `produces`, not `consumes` - https://jakarta.ee/specifications/platform/8/apidocs/javax/ws/rs/ext/messagebodywriter So I don't say you are not right, but I'm not sure if this is expected behavior.

Reply:
added` @Produces("application/yaml")` to the provider to be more consistent. Thanks @michalvavrik .